userspace/libscap/engine/savefile/scap_reader.h

```diff
@@ -0,0 +1,104 @@
+/*
+Copyright (C) 2022 The Falco Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+
+*/
+
+#pragma once
+
+#include <stdio.h>
+#include <stdint.h>
+#include <stdbool.h>
+#include <stdlib.h>
+#include "scap_assert.h"
+#include "scap_zlib.h"
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+
+/**
+ * @brief Represents a reader for data in SCAP format
+ */
+typedef struct scap_reader
+{
+    /**
+     * @brief The internal state of each implementation.
+     */
+    void* handle;
+
+    /**
+     * @brief Reads at most len bytes into buf from the given reader.
+     * On success, returns the number of bytes read. On failure,
+     * returns 0 or a negative value, and error() can be used to
```

Comment:
It'd be nice if we could have some assurance on what 0 or negative means, I would assume 0 is nothing read and negative is an error. Otherwise we would always need to check `error()` to see which of these conditions was met.